ehrapy/preprocessing/_outliers.py

```diff
@@ -0,0 +1,123 @@
+from __future__ import annotations
+
+import numpy as np
+import pandas as pd
+import scipy.stats.mstats
+from anndata import AnnData
+
+
+def winsorize(
+    adata: AnnData,
+    vars: str | list[str] = None,
+    obs_cols: str | list[str] = None,
+    limits: list[float] = None,
+    copy: bool = False,
+    **kwargs,
+) -> AnnData:
+    """Returns a Winsorized version of the input array.
+
+    The implementation is based on https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.mstats.winsorize.html
+
+    Args:
+        adata: AnnData object to winsorize
+        vars: The features to winsorize
+        obs_cols: Columns in obs with features to winsorize
+        limits: Tuple of the percentages to cut on each side of the array as floats between 0. and 1. (default: 0.01 and 0.99)
+        copy: Whether to return a copy or not
+        **kwargs: Keywords arguments get passed to scipy.stats.mstats.winsorize
+
+    Returns:
+        Winsorized AnnData object if copy is True
+    """
+    if vars is not None and not all(elem in adata.var_names.values for elem in vars):
+        raise ValueError(
+            f"Columns `{[col for col in vars if col not in adata.var_names.values]}` are not in var_names."
```

Comment:
Does this work for a single string as well; because we're iterating over `vars` and if this is a `str` we're iterating over the single letters in the `str` don't we?

this could be simplified anyways by making vars a set: Something like the following should work
```
if vars is not None:
    diff = vars - set(adata.var_names)
    if diff:  # difference is not an empty set -> vars has some values that are not in the var_names
        raise ValueError(
            f"Columns `{",".join(var for var in diff)}` are not in var_names."
```

Reply:
Nice one